common/types/message/multi_auth_msg.go

```diff
@@ -0,0 +1,95 @@
+package message
+
+import (
+	"crypto/ecdsa"
+
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/common/hexutil"
+	"github.com/scroll-tech/go-ethereum/crypto"
+	"github.com/scroll-tech/go-ethereum/rlp"
+)
+
+// MultiAuthMsg is the first message exchanged from the Prover to the Sequencer.
+// It effectively acts as a registration, and makes the Prover identification
+// known to the Sequencer.
+type MultiAuthMsg struct {
+	// Message fields
+	Identity *MultiIdentity `json:"message"`
+	// Prover signature
+	Signature string `json:"signature"`
+}
+
+// AcceptVersion contains hard_fork_name and prover_version
+type AcceptVersion struct {
+	ProverVersion string `json:"prover_version"`
+	HardForkName  string `json:"hard_fork_name"`
+}
+
+// MultiIdentity contains all the fields to be signed by the prover.
```

Comment:
I think don't need add a new msg type. just remove the auth msg, and rename `legacy_auth_msg` to `auth_msg`

Reply:
`HardForkName` never needs it. and only pass maximum `proverVersion` to coordinator is ok.